Update 2024-02-monthly.md (review) (#2)
cardano-scaling/website: monthly/2024-02-monthly.md

![Fanout after contest bug sketching](./img/2024-02-fanout-after-contest.png)

While reviewing the code, the team discovered that the `Contest` observation was not extending the deadline after the contestation was made. As a result, the `Fanout` transaction was invalid because the validator checks that the `Fanout` transaction's lower validity bound is higher than the deadline to contest. This check ensures that one can only fan out after the contestation period is over.

Before fixing this bug, the team tried reproducing it using the MBT (Model Based Tests) framework, one of the tools in the test suite. However, to do that, they needed to complete modeling the entire head lifecycle and the specific scenario they observed in real life. Arbitrary actions in all head states did not reveal this bug.

> Finally, the team was able to fix the bug easily. However, reproducing the bug helped them gain more knowledge on using the framework correctly and improve their specifications.

Reviewer: ```suggestion
After also improving the framework itself to correctly validate transactions and a fix for simulating rollbacks, the team was able to fix the bug easily. However, it proves that being able to reproduce a bug is not only improving the code, but also allowed the team to gain knowledge and improve their test suites.
```

> This is the first joint report of the 🐲 Hydra and 🛡 Mithril projects and summarizes on the work done since January 2024.

Reviewer 2: ```suggestion
This is the first joint report of the 🐲 Hydra and 🛡 Mithril projects, which summarizes the teams' progress since January 2024.
```

Previously, both teams had been releasing progress updates in different formats and intervals within their respective repositories:

- Hydra [monthly reports](https://hydra.family/head-protocol/monthly)
- Mithril [dev blog](https://mithril.network/doc/dev-blog/)

> Both sites are created from the individual project repositories. However, it is not ideal to keep "procedural" documents as part of the code base, since it requires special handling when building versioned, published documentation. For example, [this workflow](https://github.com/input-output-hk/hydra/blob/fad12fd7d967e5e8af4d8b832396e68bd8510e9a/.github/workflows/publish-docs.yaml#L87-L92) in the Hydra project which publishes to https://hydra.family required several attempts to work around how docusaurus websites are built. Consequently, we introduced a [dedicated repository](https://github.com/cardano-scaling/website) that builds into a [website](https://cardano-scaling.org). This not only helps in keeping the original product repositories clean of inherently dated information, but also serves as a new home for joint reports, announcements or occasional articles.

Reviewer 2: ```suggestion
Both sites are created from the individual project repositories. However, it is not ideal to keep 'procedural' documents as part of the code base, since it requires special handling when building versioned, published documentation. For example, [this workflow](https://github.com/input-output-hk/hydra/blob/fad12fd7d967e5e8af4d8b832396e68bd8510e9a/.github/workflows/publish-docs.yaml#L87-L92) in the Hydra project which publishes to https://hydra.family required several attempts to work around how Docusaurus websites are built. Consequently, we introduced a [dedicated repository](https://github.com/cardano-scaling/website) that builds into a [website](https://cardano-scaling.org). This not only helps keep the original product repositories clean of inherently dated information, but also serves as a new home for joint reports, announcements, or occasional articles.
```

> This monthly report serves as preparation for the monthly stakeholder review meeting (see [slides][slides] and [recording][recording]). At the meeting, the team updates stakeholders on recent developments and gathers feedback on proposed plans.

Reviewer 2: ```suggestion
This monthly report serves as preparation for the monthly stakeholder review meeting (see [slides][slides] and [recording][recording]). At the meeting, the teams update stakeholders on recent developments and gather feedback on proposed plans.
```
